### ISSUE TITLE
Warning for deprecated method Object#=~

### ISSUE THREAD

**Issue by lanzhiheng** [formtastic/formtastic#1314]
```
/Users/lan/.rbenv/versions/2.7.1/lib/ruby/gems/2.7.0/gems/formtastic-3.1.5/lib/formtastic/localizer.rb:118: warning: deprecated Object#=~ is called on FalseClass; it always returns nil
```

Got some warning in ruby 2.7.1, because we used `=~` for matching. This method is deprecated. https://ruby-doc.org/core-2.7.1/Object.html#method-i-3D-7E

**Comment by raresroca05** [comment]
Will this be merged soon?

**Comment by mikz** [comment]
@raresroca05 you can see my review on both PRs. They are missing a test case and possibly can trigger an exception.

**Comment by ndbroadbent** [comment]
Is everyone seeing this warning for ActiveAdmin? Or are there some other cases where `false` is being passed in? We are looking into this issue now and will try to add a test case for this, but it might actually be a bug in ActiveAdmin.

**Comment by deivid-rodriguez** [comment]
None of our current tests currently show this warning. Otherwise, having an issue way to reproduce it, I'd come here and try to add a proper test. That doesn't mean this is not a bug in activeadmin, but I thought it could be useful information.

**Comment by pctj101** [comment]
Looks like you're looking for people with this problem? We have it so we are monitoring this issue.  If you're having trouble replicating, I'll see if I can make an example over the next few days.

**Comment by deivid-rodriguez** [comment]
Yeah, I mean, as per @mikz's comments, we're also looking for a regression spec and making sure the proposed fixes won't make any cases crash. But having a way to reproduce it also unblocks us to have a look at it ourselves.

**Comment by pctj101** [comment]
Hope this is of some help

https://github.com/pctj101/formtastic_deprecation_demo

chruby 2.7.1

mysqldump in ft.sql

uname -a -> Darwin dev.local 19.6.0 Darwin Kernel Version 19.6.0: Mon Aug 31 22:12:52 PDT 2020; root:xnu-6153.141.2~1/RELEASE_X86_64 x86_64

**Comment by gat-bryszard** [comment]
We're also waiting for the fix. Thanks in advance.

**Comment by deivid-rodriguez** [comment]
I managed to repro with @pctj101's app on my system. I'll try to take a look soon.

**Comment by deivid-rodriguez** [comment]
Actually, I can't repro this with formtastic 4.0.0.rc1. Could you double check? You can opt-in to the prerelease by adding `gem "formtastic", "~> 4.0.a` to your `Gemfile` and running `bundle update formtastic`.

**Comment by bryszard** [comment]
@deivid-rodriguez I confirm, it is fixed in 4.0.0.rc1.

One doubt here is - when can we expect 4.0.0? Is 4.0.0.rc1 a stable version that we can use in production?

**Comment by deivid-rodriguez** [comment]
I think given that it's been more than 3 years since the previous stable release, many people has been pointing to the default branch in github since. So, yeah, I'd say it's fairly stable.

That said, the more people try it out, the more issues/regressions will get reported and the more likely it will be that the final version is released. It's the classic chicken and egg issue with prereleases: people won't test them out because of not being stable, and they are not stable because people won't test them out and report issues :man_shrugging:.

**Comment by bryszard** [comment]
We'll do the leap of faith then and report if we see any issue :). Thanks for the explanation.

**Comment by dmitrytrager** [comment]
Version 4 can have major changes and be incompatible with version 3.
I believe all warnings for version 3 should be fixed within version 3.

**Comment by justinfrench** [comment]
Thanks for this everyone.

> I believe all warnings for version 3 should be fixed within version 3.

Agree. I've merged #1366 into the 3.1-stable branch. Not sure how many people still use it, but I'll try to push a new 3.x gem from that branch.

**Comment by justinfrench** [comment]
Built & pushed: https://rubygems.org/gems/formtastic/versions/3.1.6

Closing this, please file a new issue if there's anything wrong with the new gem.
